app/scripts/controllers/user-storage/encryption.ts

```diff
@@ -23,14 +36,57 @@ function bytesToUtf8(byteArray: Uint8Array) {
   return decoder.decode(byteArray);
 }
 
+/**
+ * In Memory Cache of the derived key from a given salt.
+ * We can tidy this to make it reusable with future versions and KDF functions
+ */
+const inMemCachedKDF: Map<string, Uint8Array> = new Map();
```

Comment:
I've inlined this in-mem cache. Meh we can refactor this into a separate file as part of the Shared Libraries work.